task3/Task3/Domain/Meal.cs

```diff
@@ -0,0 +1,43 @@
+﻿
+namespace Task3
+{
+    class Meal
+    {
+        public string Name { get; set; }
+        public double Price { get; set; }
+        public string ImagePath { get; set; }
+
+        private Meal(string name, double price)
+        {
+            if (name == null)
+            {
+                throw new System.ArgumentNullException("Parameter cannot be null", "name");
+            }
```

Comment:
Wrong usage of `ArgumentNullException`
Check documentation for it:
> public ArgumentNullException (string paramName, string message);

The parameter name should go first, then goes the error message